File: analytic-engine/src/main/java/com/mycompany/models/JobModel.java
package com.mycompany.models;

import java.util.Objects;

public class JobModel {
    public String _id;
    public String __v;
    public String name;
    public Integer jobStatus;
    public String description;
    public String userId;
    public String rawInputDirectory;
    public String stagingFileName;
    public String createdAt;
    public String updatedAt;
    public boolean generateESIndices;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        JobModel jobModel = (JobModel) o;
        return generateESIndices == jobModel.generateESIndices &&
                Objects.equals(_id, jobModel._id) &&
                Objects.equals(__v, jobModel.__v) &&
                Objects.equals(name, jobModel.name) &&
                Objects.equals(jobStatus, jobModel.jobStatus) &&
                Objects.equals(description, jobModel.description) &&
                Objects.equals(userId, jobModel.userId) &&
                Objects.equals(rawInputDirectory, jobModel.rawInputDirectory) &&
                Objects.equals(stagingFileName, jobModel.stagingFileName) &&
                Objects.equals(createdAt, jobModel.createdAt) &&
                Objects.equals(updatedAt, jobModel.updatedAt);
    }

    @Override
    public int hashCode() {
        return Objects.hash(_id, __v, name, jobStatus, description, userId, rawInputDirectory, stagingFileName, createdAt, updatedAt, generateESIndices);
    }
}

File: mongodb-service/src/models/jobModel.ts
import * as mongoose from "mongoose";
import { Model } from "mongoose";

/**
 * Job type, model mongoose document and schema definition. Models are used by
 * Controllers so they are picked up by the swagger. Documents and schemas are used by 
 * Repositories so model defined Mongodb functions can be used.
 */
type JobType = IJob & mongoose.Document;
export interface IJobModel {
    _id?: string;
    name: string;
    description: string;
    rawInputDirectory: string;
    stagingFileName: string;
    userId: string;
    generateESIndices: boolean;
    jobStatus: number;
}

export interface IJob extends IJobModel, mongoose.Document {
    _id: string;
}

export const JobSchema = new mongoose.Schema({
    description: {
        required: true,
        type: String
    },
    name: {
        required: true,
        type: String
    },
    generateESIndices: {
        default: true,
        required: true,
        type: Boolean
    },
    rawInputDirectory: {
        required: false,
        type: String
    },
    stagingFileName: {
        required: false,
        type: String
    },
    jobStatus: {
        default: 0,
        required: true,
        type: Number
    },
    userId: {
        required: true,
        type: String
    }
}, { timestamps: true });

const Job: Model<IJob> = mongoose.model<JobType>("Job", JobSchema);
export default Job;
